Monika After Story/game/zz_selector.rpy

```diff
@@ -3291,6 +3439,39 @@ label mas_selector_sidebar_select(items, select_type, preview_selections=True, o
                 mailbox
             )
 
+        # make filter_map items MASSelectableImageButtonDisplayable
+        if filter_map:
+            filter_map_selbtn = {}
+            # only add unlock
+            if only_unlocked:
+                for filter_items in filter_map:
+                    filter_map_selbtn[filter_items] = [
+                        MASSelectableImageButtonDisplayable(
+                            item,
+                            select_map,
+                            viewport_bounds,
+                            mailbox,
+                            False, # TODO: multi-select
+                            item.disable_type
+                        )
+                        for item in filter_map[filter_items]
+                        if item.unlocked
+                    ]
+            else:
+                for filter_items in filter_map:
+                    filter_map_selbtn[filter_items] = [
+                        MASSelectableImageButtonDisplayable(
+                            item,
+                            select_map,
+                            viewport_bounds,
+                            mailbox,
+                            False, # TODO: multi-select
+                            item.disable_type
+                        )
+                        for item in filter_map[filter_items]
+                    ]
+            filter_map = filter_map_selbtn
+
```

Comment:
this should use the same selectable displayables as the ones in `disp_items`. (So you should loop over the `disp_items` for each selectable in `filter_map` to get the displayables after `disp_items` is set) 

Otherwise there could be weirdness since this currently results in two displayables per selectable.

Reply:
I have done a new commit that should fix this